lib/apipie/generator/swagger/param_description/builder.rb

```diff
@@ -98,8 +98,8 @@ def required_from_path?
   def warn_optional_without_default_value(definition)
     if !required? && !definition.key?(:default)
       method_id =
-        if @param_description.is_a?(Apipie::ResponseDescriptionAdapter::PropDesc)
-          @controller_method
+        if @controller_method.present? && @param_description.is_a?(Apipie::ResponseDescriptionAdapter::PropDesc)
```

Comment:
Can we end up with a blank `@controller_method` ? 
If so maybe we could add a test case for that, update the builder's YARD block to

```ruby
# @param [Apipie::MethodDescription, nil] controller_method
```

Reply:
I admit that I haven't _really_ dug into the "could `@controller_method` be `nil`" question.  I added the `@controller_method.present?` check because of a test failing.  I don't _really_ know if in the normal usage of these objects to define API docs you would ever have a `nil` in there.  But, obviously, the code doesn't actually prevent it.

I could take the check out and try to trace the test that failed a bit deeper to understand why it's able to construct the world where it's exercising these objects without a `@controller_method` being set.  I don't really know what else might break elsewhere if it's `nil`.

Reply:
A `nil` `@controller_method` is expected if we call `json_schema_for_self_describing_class` (see https://github.com/Apipie/apipie-rails/blob/4be37413e447a39dd7c117c5c6fd8fad41d3ddf9/lib/apipie/swagger_generator.rb#L36-L43).  I'll update the YARD block and introduce an explicit test for it being `nil`.

Reply:
Ok, added a new commit that changes the YARD for `Apipie::Generator::Swagger::ParamDescription::Builder` and puts in tests for both Apipie::Generator::Swagger::ParamDescription::Builder and Apipie::Generator::Swagger::ParamDescription::Type when emitting warnings with `PropDesc` for a `param_description` and `nil` for a `controller_method`.